analyze_blog_posts.py

```
@@ -0,0 +1,188 @@
+            for fmt in ['%Y-%m-%d %H:%M:%S', '%Y-%m-%d', '%Y-%m-%d %H:%M:%S.%f']:
+                try:
+                    date_obj = datetime.strptime(date_str, fmt)
+                    break
+                except ValueError:
+                    continue
+            else:
+                # Try parsing with dateutil if available
+                try:
+                    from dateutil.parser import parse
+                    date_obj = parse(date_str)
+                except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)